### PR TITLE
ENC: add multiset estimate (bayes_estimate deprecated)

### PR DESCRIPTION
**Pull request by nghi-truyen**
- Implement multiset estimate (bayes_estimate is deprecated)
- Backup cost_variant and cost_options while calling multiple optimize that will be used for multiset estimate
- Refactorize standardize args of multiple optimize for checking if mapping == "ann"
- Add test for multiset estimate

New features:
- resolving memory problem when running on large dataset. It uses now parallel Fortran-based multiple simulation
- ability to estimate at inactive cell in case of regionalization
- possibility to select parameters for estimating which are not necessary to be consistent with optimize_options["parameters"]
- adding joint-probability computation.

### CODE REVIEW

File: smash/core/simulation/estimate/estimate.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+)
+
+from smash.core.simulation.optimize.optimize import MultipleOptimize
+from smash.core.simulation.run.run import MultipleForwardRun
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from smash._typing import Numeric, ListLike
+    from smash.core.model.model import Model
```
Comment (inoelloc) [inline smash/core/simulation/estimate/estimate.py:16]: En dessous de numpy

File: smash/core/simulation/optimize/optimize.py [hunk 2]
```diff
@@ -340,19 +345,51 @@ def _multiple_optimize(
...
+
+            elif op in model.opr_initial_states.keys:
+                value = model.get_opr_initial_states(op)[0, 0]
+
+            # % In case we have other kind of parameters. Should be unreachable.
+            else:
+                pass
+
+            setattr(samples_fnl, op, value * np.ones(samples.n_sample))
+            setattr(samples_fnl, "_dst_" + op, np.ones(samples.n_sample))
```
Comment (inoelloc) [inline smash/core/simulation/optimize/optimize.py:387]: Ca devrait broadcast directement ? 
```suggestion
            setattr(samples_fnl, op, value)
            setattr(samples_fnl, "_dst_" + op, 1)
```

**Review by inoelloc — APPROVED** [review]

File: smash/core/simulation/optimize/optimize.py (hunk 2 above)
Comment (inoelloc) [reply smash/core/simulation/optimize/optimize.py:387]: J'ai rien dit, ce sont des keys qui n'existent pas

File: smash/core/simulation/estimate/_tools.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from smash._typing import AnyTuple
+    from smash.core.model.model import Model
+    from smash.factory.samples.samples import Samples
+
+from tqdm import tqdm
+import numpy as np
+from scipy.stats import gaussian_kde
```
Comment (nghi-truyen) [inline smash/core/simulation/estimate/_tools.py:12]:
```suggestion
from tqdm import tqdm
import numpy as np
from scipy.stats import gaussian_kde

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from smash._typing import AnyTuple
    from smash.core.model.model import Model
    from smash.factory.samples.samples import Samples
```

**Pull request merged**